### PR TITLE
Feature/widget factory

### PR DESCRIPTION
**Pull request by tkloczko**
cf issue #40 for more details

### CODE REVIEW

File: src/dtkCore/dtkCorePluginWidgetManager.cpp [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    d->widgets.insert(ptr, w);
+    d->variants.insert(ptr, v);
+
+    emit added(v, w);
+}
+
+int dtkCorePluginWidgetManager::remove(const void *ptr)
+{
+    return d->widgets.remove(ptr);
+    return d->variants.remove(ptr);
```
Comment (nniclausse) [inline src/dtkCore/dtkCorePluginWidgetManager.cpp]: We should remove both and returns true only if both remove succeeds

File: src/dtkCore/dtkCorePluginWidgetManager.cpp [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        d->widgets.remove(k);
+        d->variants.remove(k);
+    }
+    return keys.size();
+}
+
+void dtkCorePluginWidgetManager::clear(void)
+{
+    d->widgets.clear();
+}
```
Comment (nniclausse) [inline src/dtkCore/dtkCorePluginWidgetManager.cpp]: Why only d->widgets is cleared ? why not d->variants ?

File: src/dtkWidgets/dtkWidgets [hunk 3]
```diff
@@ -30,7 +30,6 @@
 #include "dtkViewManager.h"
 #include "dtkViewWidget.h"
 #include "dtkWidgetsLogView.h"
-#include "dtkWidgetFactory.h"
```
Comment (nniclausse) [inline src/dtkWidgets/dtkWidgets]: dtkWidgetFactory.(h|tpp) are removed from the CMakeLists but the files are not removed, is it normal ?

File: src/dtkCore/dtkCorePluginWidgetManager.cpp (hunk 1 above)
Comment (tkloczko) [reply src/dtkCore/dtkCorePluginWidgetManager.cpp]: Pff, owing to both maps are updates together, we should remove entries in both but we can return the result from only one.

File: src/dtkCore/dtkCorePluginWidgetManager.cpp (hunk 2 above)
Comment (tkloczko) [reply src/dtkCore/dtkCorePluginWidgetManager.cpp]: Yes, variants map should be cleared too.

File: src/dtkWidgets/dtkWidgets (hunk 3 above)
Comment (tkloczko) [reply src/dtkWidgets/dtkWidgets]: Argh. I forgot :-(

**Pull request merged**